### ISSUE TITLE
loss, flu_loss, c_loss_1 and 2 seem `nan`

### ISSUE THREAD

**Issue by vietvo89** [Yu-Fangxu/COLD-Attack#19]
Hi I tried to reduce the batch size and ran your code. It seems the loss is weird, as all the losses are `nan` after the first iteration. 
I printed all the losses for every iteration (not just under the condition where you also print the output from the model). Is it a bug? 

```python
1, loss: -945.2872,flu_loss: 7.1676, c_loss_1: 2.7090, c_loss_2: 13.6562, lr: 0.1000, |.                   . |
iter=1, loss: -945.2872,flu_loss: 7.1676, c_loss_1: 2.7090, c_loss_2: 13.6562, lr: 0.1000
iter=2, loss: nan,flu_loss: nan, c_loss_1: nan, c_loss_2: nan, lr: 0.1000
iter=3, loss: nan,flu_loss: nan, c_loss_1: nan, c_loss_2: nan, lr: 0.1000
iter=4, loss: nan,flu_loss: nan, c_loss_1: nan, c_loss_2: nan, lr: 0.1000
iter=5, loss: nan,flu_loss: nan, c_loss_1: nan, c_loss_2: nan, lr: 0.1000
iter=6, loss: nan,flu_loss: nan, c_loss_1: nan, c_loss_2: nan, lr: 0.1000
iter=7, loss: nan,flu_loss: nan, c_loss_1: nan, c_loss_2: nan, lr: 0.1000
iter=8, loss: nan,flu_loss: nan, c_loss_1: nan, c_loss_2: nan, lr: 0.1000
iter=9, loss: nan,flu_loss: nan, c_loss_1: nan, c_loss_2: nan, lr: 0.1000
iter=10, loss: nan,flu_loss: nan, c_loss_1: nan, c_loss_2: nan, lr: 0.1000
iter=11, loss: nan,flu_loss: nan, c_loss_1: nan, c_loss_2: nan, lr: 0.1000
iter=12, loss: nan,flu_loss: nan, c_loss_1: nan, c_loss_2: nan, lr: 0.1000
iter=13, loss: nan,flu_loss: nan, c_loss_1: nan, c_loss_2: nan, lr: 0.1000
...
```

**Comment by vietvo89** [comment]
it seems the `epsilon` does not update properly! How to solve it?

```python
Parameter containing:
tensor([[[    nan,     nan,     inf,  ...,     nan,     nan,     nan],
         [    nan,     nan,  0.1056,  ...,     nan,     nan,     nan],
         [    nan,  0.0849,     inf,  ...,     inf,     nan,     nan],
         ...,
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan]],

        [[    nan,     nan,     inf,  ...,     nan,     nan,     nan],
         [    nan,     nan,  0.1006,  ...,     nan,     nan,     nan],
         [    nan, -0.0980,     inf,  ...,     inf,     nan,     nan],
         ...,
         [    nan,     nan,     inf,  ...,     nan,     nan,     nan],
         [    nan,     nan,    -inf,  ...,     nan,     nan,     nan],
         [    nan,    -inf,  0.1000,  ...,     nan,     nan,     nan]],

        [[    nan,     nan,     inf,  ...,     nan,     nan,     nan],
         [    nan,     nan,  0.1075,  ...,     nan,     nan,     nan],
         [    nan,     inf,     inf,  ...,     inf,     nan,     nan],
         ...,
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan]],

        ...,

        [[    nan,     nan,    -inf,  ...,     nan,     nan,     nan],
         [    nan,     nan,  0.1084,  ...,     nan,     nan,     nan],
         [    nan,     inf,     inf,  ...,     inf,     nan,     nan],
         ...,
         [    nan,     nan,     inf,  ...,     nan,     nan,     nan],
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan]],

        [[    nan,     nan,     inf,  ...,     nan,     nan,     nan],
         [    nan,     nan,  0.1190,  ...,     nan,     nan,     nan],
         [    nan, -0.1007,     inf,  ...,     inf,     nan,     nan],
         ...,
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan]],

        [[    nan,     nan,     inf,  ...,     nan,     nan,     nan],
         [    nan,     nan,  0.1063,  ...,     nan,     nan,     nan],
         [    nan,     inf,     inf,  ...,     inf,     nan,     nan],
         ...,
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan]]],
       device='cuda:0', dtype=torch.float16, requires_grad=True)
```

**Comment by danshi777** [comment]
Hi, I also encountered the same problem when running `attack.sh "suffix"`. The losses in the subsequent iterations would all become `nan`. Have you solved it?

**Comment by vietvo89** [comment]
@danshi777 , I have no idea how to fix it. @Yu-Fangxu , could you address this problem? Is there any bug or environment problems?

**Comment by vietvo89** [comment]
@danshi777, did you have the same issue as reported in https://github.com/Yu-Fangxu/COLD-Attack/issues/18 ? it seems the transformers==4.43.1 might help.
